// @aparajita/capacitor-biometric-auth/dist/esm/native.js
import { BiometricAuthBase } from './base';
import { BiometryErrorType, BiometryType } from './definitions';
// eslint-disable-next-line import/prefer-default-export
export class BiometricAuthNative extends BiometricAuthBase {
    constructor(capProxy) {
        super();
        /*
          In order to call native methods and maintain the ability to
          call pure Javascript methods as well, we have to bind the native methods
          to the proxy.
    
          capProxy is a proxy of an instance of this class, so it is safe
          to cast it to this class.
        */
        // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
        const proxy = capProxy;
        /* eslint-disable @typescript-eslint/unbound-method */
        this.checkBiometry = proxy.checkBiometry;
        this.internalAuthenticate = proxy.internalAuthenticate;
        /* eslint-enable */
    }
    // @native
    async checkBiometry() {
        // Never used, but we have to satisfy the compiler.
        return Promise.resolve({
            isAvailable: true,
            biometryType: BiometryType.none,
            biometryTypes: [],
            reason: '',
            code: BiometryErrorType.none,
        });
    }
    // @native
    // On native platforms, this will present the native authentication UI.
    async internalAuthenticate(
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    options) { }
    // Web only, used for simulating biometric authentication.
    // eslint-disable-next-line @typescript-eslint/require-await
    async setBiometryType(
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    type) {
        throw this.unimplemented('setBiometryType() is web only');
    }
}
