// @aparajita/capacitor-biometric-auth/dist/plugin.js
var capacitorBiometricAuth = (function (exports, core, app) {
    'use strict';

    // noinspection JSUnusedGlobalSymbols
    exports.BiometryType = void 0;
    (function (BiometryType) {
        /**
         * No biometry is available
         */
        BiometryType[BiometryType["none"] = 0] = "none";
        /**
         * iOS Touch ID is available
         */
        BiometryType[BiometryType["touchId"] = 1] = "touchId";
        /**
         * iOS Face ID is available
         */
        BiometryType[BiometryType["faceId"] = 2] = "faceId";
        /**
         * Android fingerprint authentication is available
         */
        BiometryType[BiometryType["fingerprintAuthentication"] = 3] = "fingerprintAuthentication";
        /**
         * Android face authentication is available
         */
        BiometryType[BiometryType["faceAuthentication"] = 4] = "faceAuthentication";
        /**
         * Android iris authentication is available
         */
        BiometryType[BiometryType["irisAuthentication"] = 5] = "irisAuthentication";
    })(exports.BiometryType || (exports.BiometryType = {}));
    /**
     * If the `authenticate()` method throws an exception, the `BiometryError`
     * instance contains a .code property which will contain one of these strings,
     * indicating what the error was.
     *
     * See https://developer.apple.com/documentation/localauthentication/laerror
     * for a description of each error code.
     */
    exports.BiometryErrorType = void 0;
    (function (BiometryErrorType) {
        BiometryErrorType["none"] = "";
        BiometryErrorType["appCancel"] = "appCancel";
        BiometryErrorType["authenticationFailed"] = "authenticationFailed";
        BiometryErrorType["invalidContext"] = "invalidContext";
        BiometryErrorType["notInteractive"] = "notInteractive";
        BiometryErrorType["passcodeNotSet"] = "passcodeNotSet";
        BiometryErrorType["systemCancel"] = "systemCancel";
        BiometryErrorType["userCancel"] = "userCancel";
        BiometryErrorType["userFallback"] = "userFallback";
        BiometryErrorType["biometryLockout"] = "biometryLockout";
        BiometryErrorType["biometryNotAvailable"] = "biometryNotAvailable";
        BiometryErrorType["biometryNotEnrolled"] = "biometryNotEnrolled";
        BiometryErrorType["noDeviceCredential"] = "noDeviceCredential";
    })(exports.BiometryErrorType || (exports.BiometryErrorType = {}));
    /**
     * `authenticate()` throws instances of this class.
     */
    class BiometryError {
        constructor(message, code) {
            this.message = message;
            this.code = code;
        }
    }

    const kBiometryTypeNameMap = {
        [exports.BiometryType.none]: '',
        [exports.BiometryType.touchId]: 'Touch ID',
        [exports.BiometryType.faceId]: 'Face ID',
        [exports.BiometryType.fingerprintAuthentication]: 'Fingerprint Authentication',
        [exports.BiometryType.faceAuthentication]: 'Face Authentication',
        [exports.BiometryType.irisAuthentication]: 'Iris Authentication',
    };
    /**
     * Return a human-readable name for a BiometryType.
     */
    // eslint-disable-next-line import/prefer-default-export
    function getBiometryName(type) {
        return kBiometryTypeNameMap[type] || '';
    }

    const proxy = core.registerPlugin('BiometricAuthNative', {
        web: async () => Promise.resolve().then(function () { return web; }).then((module) => new module.BiometricAuthWeb()),
        ios: async () => Promise.resolve().then(function () { return native; }).then((module) => new module.BiometricAuthNative(proxy)),
        android: async () => Promise.resolve().then(function () { return native; }).then((module) => new module.BiometricAuthNative(proxy)),
    });

    // eslint-disable-next-line import/prefer-default-export
    class BiometricAuthBase extends core.WebPlugin {
        async authenticate(options) {
            try {
                await this.internalAuthenticate(options);
            }
            catch (error) {
                // error will be an instance of CapacitorException on native platforms,
                // an instance of BiometryError on the web.
                if (error instanceof core.CapacitorException) {
                    throw new BiometryError(error.message, 
                    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- we are converting from ExceptionCode to BiometryErrorType
                    error.code);
                }
                else {
                    throw error;
                }
            }
        }
        addResumeListener(listener) {
            return app.App.addListener('appStateChange', ({ isActive }) => {
                if (isActive) {
                    this.checkBiometry()
                        .then((info) => {
                        listener(info);
                    })
                        .catch(console.error);
                }
            });
        }
    }

    // eslint-disable-next-line import/prefer-default-export
    class BiometricAuthWeb extends BiometricAuthBase {
        constructor() {
            super(...arguments);
            this.biometryType = exports.BiometryType.none;
        }
        // On the web, return the fake biometry set by setBiometryType().
        async checkBiometry() {
            return Promise.resolve({
                isAvailable: this.biometryType !== exports.BiometryType.none,
                biometryType: this.biometryType,
                biometryTypes: [],
                reason: '',
                code: exports.BiometryErrorType.none,
            });
        }
        // On the web, fake authentication with a confirm dialog.
        async internalAuthenticate(options) {
            const { isAvailable, biometryType } = await this.checkBiometry();
            if (isAvailable) {
                if (
                // eslint-disable-next-line no-alert
                confirm(
                // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing -- we want to use the default value if options?.reason is an empty string
                (options === null || options === void 0 ? void 0 : options.reason) ||
                    `Authenticate with ${getBiometryName(biometryType)}?`)) {
                    return;
                }
                throw new BiometryError('User cancelled', exports.BiometryErrorType.userCancel);
            }
            throw new BiometryError('Biometry not available', exports.BiometryErrorType.biometryNotAvailable);
        }
        // Web only, used for simulating biometric authentication.
        async setBiometryType(type) {
            if (typeof type === 'undefined') {
                return Promise.resolve();
            }
            if (typeof type === 'string') {
                // eslint-disable-next-line no-prototype-builtins
                if (exports.BiometryType.hasOwnProperty(type)) {
                    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
                    this.biometryType = exports.BiometryType[type];
                }
            }
            else {
                this.biometryType = type;
            }
            return Promise.resolve();
        }
    }

    var web = /*#__PURE__*/Object.freeze({
        __proto__: null,
        BiometricAuthWeb: BiometricAuthWeb
    });

    // eslint-disable-next-line import/prefer-default-export
    class BiometricAuthNative extends BiometricAuthBase {
        constructor(capProxy) {
            super();
            /*
              In order to call native methods and maintain the ability to
              call pure Javascript methods as well, we have to bind the native methods
              to the proxy.
        
              capProxy is a proxy of an instance of this class, so it is safe
              to cast it to this class.
            */
            // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
            const proxy = capProxy;
            /* eslint-disable @typescript-eslint/unbound-method */
            this.checkBiometry = proxy.checkBiometry;
            this.internalAuthenticate = proxy.internalAuthenticate;
            /* eslint-enable */
        }
        // @native
        async checkBiometry() {
            // Never used, but we have to satisfy the compiler.
            return Promise.resolve({
                isAvailable: true,
                biometryType: exports.BiometryType.none,
                biometryTypes: [],
                reason: '',
                code: exports.BiometryErrorType.none,
            });
        }
        // @native
        // On native platforms, this will present the native authentication UI.
        async internalAuthenticate(
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        options) { }
        // Web only, used for simulating biometric authentication.
        // eslint-disable-next-line @typescript-eslint/require-await
        async setBiometryType(
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        type) {
            throw this.unimplemented('setBiometryType() is web only');
        }
    }

    var native = /*#__PURE__*/Object.freeze({
        __proto__: null,
        BiometricAuthNative: BiometricAuthNative
    });

    exports.BiometricAuth = proxy;
    exports.BiometryError = BiometryError;
    exports.getBiometryName = getBiometryName;

    return exports;

})({}, capacitorExports, app);
